src/Magento/FunctionalTestingFramework/Util/ModuleResolver.php

```diff
@@ -575,4 +839,43 @@ private function getBackendUrl()
     {
         return getenv('MAGENTO_BACKEND_BASE_URL') ?: getenv('MAGENTO_BASE_URL');
     }
+
+    /**
+     * Find vendor and module name from path
+     *
+     * @param string $path
+     * @return string
+     */
+    private function findVendorAndModuleNameFromPath($path)
+    {
+        $path = str_replace(DIRECTORY_SEPARATOR . self::TEST_MFTF_PATTERN, '', $path);
+        return $this->findVendorNameFromPath($path) . '_' . basename($path);
+    }
+
+    /**
+     * Find vendor name from path
+     *
+     * @param string $path
+     * @return string
+     */
+    private function findVendorNameFromPath($path)
+    {
+        $possibleVendorName = 'UnknownVendor';
+        $dirPaths = [
+            self::VENDOR,
+            self::APP_CODE,
+            self::DEV_TESTS
+        ];
+
+        foreach ($dirPaths as $dirPath) {
+            $regex = "~.+\\/" . $dirPath . "\/(?<" . self::VENDOR . ">[^\/]+)\/.+~";
```

Comment:
Small refactor for legibility:
```
            $regex = '~/' . $dirPath . '/([^/]+)~';
            // Escape directory separators
            $regex = str_replace('/', '\/', $regex);
            preg_match($regex, $path, $match);
            if (isset($match[1])) {
                $possibleVendorName = ucfirst($match[1]);
                break;
            }
```

What do you think?

Reply:
Let's discuss in person